### PR TITLE
Refactor System.IO.FileSystem tests

### PR DESCRIPTION
**Pull request by Maxwe11**
Begins progress against #1412. Also partially fixes issues #915 and #516.
Primary changes:
- Renamed EnumerableUtils -> TestFileSystemEntries and implemented `IDisposable`
- "TestWhileEnumerating" moved to their own classes which also implement `IDisposable`
- Classes which require shared test context implement `IClassFixture`
- Enable `File_FileEnumerableTests` which was disabled due to internals visibility
- Removed duplicate tests (e.g. argument validation tests were presented twice in Directory_Enumerable_API.cs and GetFiles_str.cs etc.)

Minor changes:
- Remove a bunch of `Console.WriteLine`
- Replace hand-rolled tests for exceptions with xunit's `Assert.Throws`
- Make tests data-driven (TheoryAttribute usage)

With this changes code coverage left on the same level.

### CODE REVIEW

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
failed due to #1434 (unrelated to this PR)

**Comment by ellismg** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

File: src/System.IO.FileSystem/tests/Directory/Directory_EnumerableAPIs.cs [hunk 1]
```diff
@@ -3,742 +3,234 @@
...
-using System.Text;
 using System.IO;
-using System.Security;
+using System.Linq;
 using Xunit;
 
 namespace EnumerableTests
 {
-    public class Directory_EnumerableTests
+    public class Directory_EnumerableTests : IClassFixture<TestFileSystemEntries>
```
Comment (weshaggard) [inline src/System.IO.FileSystem/tests/Directory/Directory_EnumerableAPIs.cs]: @FiveTimesTheFun - FYI our internal test infrastructure doesn't support IClassFixture so we will either need to add support for that or do this in a different way when run this test there.

Comment (Maxwe11) [inline src/System.IO.FileSystem/tests/Directory/Directory_EnumerableAPIs.cs]: Closing PR for now but will prepare new one)
BTW `IClassFixture` support that is really needed.

Comment (weshaggard) [inline src/System.IO.FileSystem/tests/Directory/Directory_EnumerableAPIs.cs]: Why did you close the PR?

Comment (Maxwe11) [inline src/System.IO.FileSystem/tests/Directory/Directory_EnumerableAPIs.cs]: I'm working on the final fix for #1412 and that fix doesn't rely on IClassFixture. Also I want to make PR more atomic and easier for reviewing.

**Comment by jasonwilliams200OK** [comment]
@Maxwe11, (since it is not so obvious) why was this closed?
It might have addressed an old hanging issue https://github.com/dotnet/corefx/issues/516.. :)

**Comment by stephentoub** [comment]
@jasonwilliams200OK, the FileSystem tests have already been overhauled.  Is there a specific case you're concerned about still?

**Comment by jasonwilliams200OK** [comment]
@stephentoub, thanks for the info. I just asked because I was wondering if I can contribute somehow to get some oldest opened issues get closed before v1.0 is RTM'd (targeting especially those 2-digits and 3-digits issue number). :)

**Comment by stephentoub** [comment]
>  I just asked because I was wondering if I can contribute somehow to get some oldest opened issues get closed before v1.0 is RTM'd

Thanks!

**Pull request closed without merging**